### PR TITLE
Use correct preprocessor with maps

### LINKED ISSUE #36: Mapping from Bean to Map with type hint does not use conversion/preprocessing
If I convert from a bean to a map with a type hint, the preprocessing is correctly created but not used.

### PR DESCRIPTION
**Pull request by tarioch**
Preprocessors are not correctly used with maps. In my case I had an issue where I convert into a map and provide a type hint. Fixes #36

### CODE REVIEW

File: src/test/groovy/org/nomin/functional/MapingToMapTest.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def nomin = new Nomin(Animal2Map)
+
+  def animal = new Animal(name: '42')
+
+  @org.junit.Test
+  void testAnimal2Map() {
+    Map map = nomin.map(animal, HashMap)
+    assert map?.size() == 1 && map['name'] == 42
+  }
+
```
Comment (dobrynya) [inline src/test/groovy/org/nomin/functional/MapingToMapTest.groovy]: Hi @tarioch 
Could you please add testing the reverse mapping like
```java
@Test
  void testMap2Animal() {
    def pet = nomin.map([name: "Pet"], Animal)
    assert pet.name == "Pet"
  }
```
?

File: src/test/groovy/org/nomin/mappings/Animal2Map.groovy [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.nomin.mappings
+
+import org.nomin.Mapping
+import org.nomin.entity.Animal
+
+/**
+ * Mapping between an animal and a simple map.
+ */
+class Animal2Map extends Mapping {
```
Comment (dobrynya) [inline src/test/groovy/org/nomin/mappings/Animal2Map.groovy]: It would be great to place this mapping into the testing class

**Comment by tarioch** [comment]
Thanks for the review and I'll update this tomorrow :)

**Comment by tarioch** [comment]
Updated the PR with your findings

**Pull request merged**